src/main/java/com/vaadin/exampledata/ChanceBinary.java

```diff
@@ -0,0 +1,23 @@
+package com.vaadin.exampledata;
+
+import java.time.LocalDateTime;
+import java.util.Random;
+
+public class ChanceBinary extends DataType<byte[]> {
+
+    private byte[][] options;
+
+    public ChanceBinary(String fileNames) {
+        this.options = FileCache.getBinaries(fileNames);

```

Comment:
Maybe not relevant but why does this have to read all files instead of selecting a random one and then reading / caching that one?

Reply:
Changed the logic to cache one file at a time.